src/beamlime/applications/_nexus_helpers.py

```diff
@@ -321,25 +334,15 @@ def _merge_f144(group: NexusGroup, data: DeserializedMessage) -> None:
         (time["config"]["values"], [data["timestamp"]])
     )
     value = cast(NexusDataset, find_nexus_structure(group, ("value",)))
-    value["config"]["values"] = np.concatenate(
-        (value["config"]["values"], data["value"])
-    )
-
-
-def _find_tdct(
-    structure: NexusStructure, data: DeserializedMessage
-) -> Iterable[NexusPath]:
-    name = data["name"]
-    for path, node in iter_nexus_structure(structure):
-        if node.get("name") == name:
-            for _, child in iter_nexus_structure(node):
-                if child.get("module") == "tdct":
-                    yield (*path, 'top_dead_center')
+    if value["config"]["values"].shape[0] == 0:  # First data
```

Comment:
Not really. If the shape is different, it just replaces the values with the coming one.

Previously we would initialize an empty array based on the shape of the incoming data piece, since we don't know the shape of the data until it arrives.

So instead of initializing the empty array as a seed, we just start with the first data piece to concatenate from.

Reply:
Hm, I see. But if the initial array is replaced anyway, why not just initialize with `None` instead?

Reply:
@jokasimr Good point. I updated it to initialize with `None` instead.